### PR TITLE
ivi-share: introduce release_shared_name request

### PR DESCRIPTION
**Pull request by kenji-hosokawa**
The compositor was unable to know whether shared buffers no longer needed.
Therefore, in spite of the consumer still using the buffer, the compositor was releasing the buffer to the producer.
In order to solve this problem, the request is introduced.

### CODE REVIEW

File: weston-ivi-shell/src/ivi-share.h [hunk 1]
```diff
@@ -44,6 +51,7 @@ struct ivi_share_nativesurface
     uint32_t send_flag;
     struct wl_listener surface_destroy_listener;
     struct ivi_shell_share_ext *shell_ext;
+    struct ivi_share_buffer_reference buffer_refs[2], *current, *back;
```
Comment (eucan) [inline weston-ivi-shell/src/ivi-share.h]: why do you need a static array and two pointers, which points contents of the array all the time? 
you can just  have two buffer members:
struct ivi_share_buffer_reference current, back;

File: weston-ivi-shell/src/ivi-share.c [hunk 2]
```diff
@@ -504,7 +504,7 @@ share_get_ivi_share_surface(struct wl_client *client, struct wl_resource *resour
 
     nativesurf = find_nativesurface(shsurf, shell_ext);
     if (nativesurf == NULL) {
-        nativesurf = alloc_share_nativesurface(shsurf->surface, id, 0,
+        nativesurf = alloc_share_nativesurface(shsurf->surface, id, shsurf->surface_id,
                                                (int32_t)IVI_SHARE_SURFACE_TYPE_GBM,
                                                (int32_t)IVI_SHARE_SURFACE_FORMAT_ARGB8888,
                                                shell_ext);
```
Comment (eucan) [inline weston-ivi-shell/src/ivi-share.c]: this patch is unrelated bug fix. please send all your bug fix patches in a separate PR.

File: protocol/ivi-share.xml [hunk 3]
```diff
@@ -135,5 +135,15 @@
...
+        <!-- Version 2 additions -->
+
+        <request name="release_shared_name" since="2">
+            <description summary="consumer releases shared name">
+              Send when the consumer no longer uses the share name notified by the damage event.
+              Until the consumer notifies the release, the compositor will not release the buffer to the producer.
+            </description>
+            <arg name="name" type="uint"/>
+        </request>
     </interface>
```
Comment (eucan) [inline protocol/ivi-share.xml]: It looks good. But you should update copyright notice to 2012-2017

File: weston-ivi-shell/src/ivi-share-gbm.c [hunk 4]
```diff
@@ -33,8 +33,6 @@
 #include "ivi-share-server-protocol.h"
 #include "ivi-share.h"
 
-static uint32_t nativesurface_name;
```
Comment (eucan) [inline weston-ivi-shell/src/ivi-share-gbm.c]: unrelated bug fixes. send in different PR.

Comment (eucan) [reply weston-ivi-shell/src/ivi-share-gbm.c]: or at least reoder your patches that preparation patches come before feature patches and not in between

File: weston-ivi-shell/src/ivi-share-gbm.c [hunk 5]
```diff
@@ -116,14 +116,14 @@ update_buffer_nativesurface(struct ivi_share_nativesurface *p_nativesurface)
...
-        weston_log("failed to import gbm_bo\n");
+        weston_log("Texture Sharing Failed to import gbm_bo\n");
         return IVI_SHAREBUFFER_INVALID;
     }
 
     struct drm_gem_flink flink = {0};
     flink.handle = gbm_bo_get_handle(bo).u32;
     if (drmIoctl(gbm_device_get_fd(backend->gbm), DRM_IOCTL_GEM_FLINK, &flink) != 0) {
-        weston_log("gem_flink: returned non-zero failed\n");
+        weston_log("Texture Sharing gem_flink: returned non-zero failed\n");
```
Comment (eucan) [inline weston-ivi-shell/src/ivi-share-gbm.c]: unrelated to the new request..

File: ivi-layermanagement-examples/simple-ivi-share/src/simple-ivi-share.c [hunk 6]
```diff
@@ -185,6 +185,7 @@ handle_share_surface_damage(void *data, struct ivi_share_surface *share_surface,
         if (window->share_buffer.eglimage != NULL) {
             pfEglDestroyImageKHR(display->egl.egldisplay,
                                  window->share_buffer.eglimage);
+            ivi_share_surface_release_shared_name(window->share_surface, window->share_buffer.name);
```
Comment (eucan) [inline ivi-layermanagement-examples/simple-ivi-share/src/simple-ivi-share.c]: Commit message of this patch is entirely missing. Please send a proper commit message which explains your implementation.

File: weston-ivi-shell/src/ivi-share.h (hunk 1 above)
Comment (kenji-hosokawa) [reply weston-ivi-shell/src/ivi-share.h]: "current" and "back" need to swap. Please find ivi-share.c L305 for the detail.
It is needed to avoid direct value copy of structure. If the copy cost can be accepted, it can be modified to your suggestion.

**Comment by kenji-hosokawa** [comment]
I will send new pull requests after I fixed the findings.
